cmd/nvidia-dra-plugin/device_state.go

```diff
@@ -390,6 +393,21 @@ func (s *DeviceState) applySharingConfig(ctx context.Context, config configapi.S
 		allocatableDevices[r.Device] = s.allocatable[r.Device]
 	}
 
+	// allow devices only with cuda compute compatility >= 7.5 as time slicing and MPS does not work with old arch
+	shareableAllocatableDevices := make(AllocatableDevices)
+	for device, deviceType := range allocatableDevices {
+		if deviceType.Gpu != nil {
```

Comment:
Does this mean that we don't timeslice MIG devices?

Reply:
In general, does it make sense to factor these checks into a function where we can better test the various combinations of options?

Reply:
these changes also need in unpreprare function